azext_iot/tests/iothub/configurations/test_iot_config_int.py

```diff
@@ -70,6 +79,31 @@ def test_edge_set_modules(self):
                 checks=[self.check("length([*])", 4)],
             )
 
+            # Apply billable edge module (requires AAD Auth)
+            # @avagraw - The billable edge modules can only be applied using user tokens (service principals are not supported)
+            if self.user["type"] == UserTypes.user.value:
+                self.cmd(
+                    self.set_cmd_auth_type(
+                        "iot edge set-modules -d {} -n {} -g {} -k '{}'".format(
+                            edge_device_ids[0], self.entity_name, self.entity_rg, edge_billable_module_path
+                        ),
+                        auth_type=AuthenticationTypeDataplane.login.value,
+                    ),
+                    checks=[self.check("length([*])", 5)],
```

Comment:
Are we able to also assert the modulesPurchase content here since we can't yet do it for a deployment?

Reply:
Unfortunately no, since API returns 204

Reply:
Ok, but our set-modules operation is a compound operation which lists the modules on the device after returning a 200. I would assume the billable module shows up in the returned list.

Reply:
Correct, it shows up in the list and I was trying to assert that by the count increment. I understand your point though, we can look at the array of deployments to find the billable module and then assert that the module contains modulesPurchase content. I will do that.

Reply:
Another option is to use module-twin show for the system module `$edgeAgent` and query the elements to support the test. Example:

![image](https://user-images.githubusercontent.com/28658112/148450114-92d66e9d-ba1a-46e9-87e4-aaadc3023e0c.png)

Reply:
This still doesn't show the modulesPurchase contents. Even querying the module directly doesn't show the purchase info (I think this is happening because the purchase info is outside the modulesContent). 

![image](https://user-images.githubusercontent.com/51140335/148459931-d97a1eb1-93e4-4e6b-98f9-55ad4a07fea4.png)

Reply:
One solution I can think of is to update the modulesContent to include the **paid-module** properties. Which can then be queried.